src/authorizer.js

```diff
@@ -90,6 +173,12 @@ class Authorizer {
       },
       context: resolver(identity, token)
     };
+
+    if (this.configuration.usagePlan) {
+      policy.policyDocument.usageIdentifierKey = await this.resolveApiKey(clientId);
```

Comment:
I would expect this code to be
```js
policy.policyDocument.usageIdentifierKey = clientId;
await this.ensureApiKey(clientId);
```

Since the input is the same as the ouput. Will also make testing easier :)

Reply:
also if we fail we should just not set a `usageIdentifierKey` so that we can maintain a current working version (at least for now).